package RemoteLegends;

option optimize_for = LITE_RUNTIME;

// Plugin: RemoteLegends

import "world_landmass.proto";
import "world_mountain_peak.proto";
import "world_region.proto";
import "world_river.proto";
import "world_underground_region.proto";

message WorldLandmassList {
    repeated dfproto.world_landmass list = 1;
}

message WorldMountainPeakList {
    repeated dfproto.world_mountain_peak list = 1;
}

message WorldRegionList {
    repeated dfproto.world_region list = 1;
}

message WorldRiverList {
    repeated dfproto.world_river list = 1;
}

message WorldUndergroundRegionList {
    repeated dfproto.world_underground_region list = 1;
}

message SingleBool {
    optional bool Value = 1;
}

message VersionInfo {
    optional string dwarf_fortress_version = 1;
    optional string dfhack_version = 2;
    optional string remote_legends_version = 3;
}

message MyListRequest {
    optional int32 list_start = 1;
    optional int32 list_end = 2;
}

message IdRequest {
    required int32 id = 1;
}
